kroxylicious-kms-provider-aws-kms/src/main/java/io/kroxylicious/kms/provider/aws/kms/AwsKmsEdekSerde.java

```diff
@@ -26,20 +26,25 @@
  * <br/>
  * The serialization structure is as follows:
  * <ol>
+ *     <li>version byte (currently always zero)</li>
  *     <li>Protobuf Unsigned varint to hold the number of bytes required to hold the UTF-8 representation of the kekRef.</li>
```

Comment:
I'm currently serialising the `KeyId` as if it were a string.   There are alternatives:

https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#key-id

*  In the single region case, the `KeyId` _appears_ to be a UUID.  So we could serialise 128 bytes.  For the multi-region key case, again it appears to UUID prefixed with a string `mrk`.    However, I see nothing in the AWS stating that KeyIds are guaranteed to be UUID, so we'd be making assumptions.

* Should we actually be serialising the ARN rather than KeyId.  This allows cross account use-cases which might be important for some users.  Again, ARNs seem to comprise the KeyId, which suggests the possibility of the optimisation we have above.

I think ultimately, the answer might be we should have the option to do any of these things driving through configurability.  I think starting with a simple string might be the place to start.